controller.js

```diff
@@ -20,4 +24,17 @@ const getAllArticles = (request, response, next) => {
     });
 };
 
-module.exports = { getAllTopics, getAllArticles };
+//This function checks if an article exists before returning either the article or an error message
+const getArticleById = (request, response, next) => {
+  const article_id = request.params.article_id;
+  fetchArticleById(article_id)
+    .then((article) => {
+      if (article) response.status(200).send({ article });
+      else response.status(404).send({ msg: "Article not found" });
```

Comment:
This should be a promise reject from the model which a error handler would then handle instead of a response in the controller.